cookbooks/aws-parallelcluster-install/resources/mysql_client/partial/_download_and_install.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+#
+# Copyright:: 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+
+property :package_installer
+
+action :download_and_install do
+  mysql_archive_url = node['cluster']['mysql']['package']['archive']
+  mysql_tar_file = "/tmp/#{node['cluster']['mysql']['package']['file-name']}"
+
+  log "Downloading MySQL packages archive from #{mysql_archive_url}"
+
+  remote_file mysql_tar_file do
+    source mysql_archive_url
+    mode '0644'
+    retries 3
+    retry_delay 5
+    not_if { ::File.exist?(mysql_tar_file) }
+  end
+
+  bash 'Install MySQL packages' do
+    user 'root'
+    group 'root'
+    cwd '/tmp'
+    code <<-MYSQL
+        set -e
+
+        EXTRACT_DIR=$(mktemp -d --tmpdir mysql.XXXXXXX)
+        tar xf "#{mysql_tar_file}" --directory "${EXTRACT_DIR}"
+        #{new_resource.package_installer} ${EXTRACT_DIR}/*
+    MYSQL
```

Comment:
Thought: I wonder if this can be done using Chef's [package](https://docs.chef.io/resources/package/) resource with source property set. I would give it a try, in a separate PR, since this is something we can reuse in multiple places, simplifying our codebase.